File: srcNIM/automatacipkg/Sample/Greeter.nim
# [ COPYRIGHT CLASE HERE ]




proc Process*(name, location: string): string =
    ## Process it to generate a sample response based on a given name and
    ## location.
    if name == "" and location == "":
        return ""
    elif name == "":
        return "stranger from " & location & "!"
    elif location == "":
        return name & "!"
    else:
        return name & " form " & location & "!"

File: srcRUST/sample/greeter.rs
// [ COPYRIGHT CLAUSE HERE ]
pub fn process<'a>(name: &'a str, location: &'a str) -> String {
    if name == "" && location == "" {
        return "".to_string();
    } else if name == "" {
        return format!("stranger from {location}!");
    } else if location == "" {
        return format!("{name}!");
    } else {
        return format!("{name} from {location}!");
    }
}
